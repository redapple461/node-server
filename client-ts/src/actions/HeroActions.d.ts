export declare const SET_MARVEL = "SET_MARVEL";
export declare const SET_DC = "SET_DC";
export declare const SET_BOTH = "SET_BOTH";
export declare const GET_DATA = "GET_DATA";
export declare const HEROES_LOADED = "HEROES_LOADED";
export declare const ADD_HERO = "ADD_HERO";
export declare const DELETE_HERO = "DELETE_HERO";
export declare const INIT_HERO = "INIT_HERO";
export declare const UPDATE_NAME = "UPDATE_NAME";
export declare const UPDATE_ADD_HERO_NAME = "UPDATE_ADD_HERO_NAME";
export declare const UPDATE_ADD_HERO_UNIVERSE = "UPDATE_ADD_HERO_UNIVERSE";
export declare const CLEAR_DETAIL_HERO = "CLEAR_DETAIL_HERO";
export declare const DETAIL_LOADED = "DETAIL_LOADED";
export declare const UPDATE_HERO = "UPDATE_HERO";
export declare const UPDATE_UNIVERSE = "UPDATE_UNIVERSE";
export declare const ADD_ASKILL = "ADD_ASKILL";
export declare const REMOVE_ASKILL = "REMOVE_ASKILL";
export declare const CLEAR_ADD_HERO = "CLEAR_ADD_HERO";
export declare const SET_EMAIL = "SET_EMAIL";
export declare const SET_PASSWORD = "SET_PASSWORD";
export declare const SET_REG_EMAIL = "SET_REG_EMAIL";
export declare const SET_REG_PASSWORD = "SET_REG_PASSWORD";
export declare const SET_USER_ID = "SET_USER_ID";
export declare const SET_JWT = "SET_JWT";
export declare const LOGOUT = "LOGOUT";
export declare const SET_USER_NAME = "SET_USER_NAME";
export declare const SET_USER_SURNAME = "SET_USER_SURNAME";
export declare const SET_USER_PHONE = "SET_USER_PHONE";
export declare const CLEAR_REG_DATA = "CLEAR_REG_DATA";
