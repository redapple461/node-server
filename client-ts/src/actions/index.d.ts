import { Hero } from '../models/Hero';
import { HeroActionsType } from '../actions/types/HeroTypes';
export declare function setMarvel(): HeroActionsType;
export declare function setDC(): HeroActionsType;
export declare function setBoth(): HeroActionsType;
export declare function getData(data: Hero[]): HeroActionsType;
export declare function loadComplete(): HeroActionsType;
export declare function addHero(hero: Hero): HeroActionsType;
export declare function deleteHero(id: number): HeroActionsType;
export declare function initDetailHero(hero: Hero): HeroActionsType;
export declare function updateName(newName: string): HeroActionsType;
export declare function updateUniverse(newUniverse: string): HeroActionsType;
export declare function updateHero(): HeroActionsType;
export declare function updateAddHeroName(addName: string): HeroActionsType;
export declare function updateAddHeroUniverse(addUniverse: string): HeroActionsType;
export declare function clearDetailHero(): HeroActionsType;
export declare function isLoad(): HeroActionsType;
export declare function addSkill(skillToAdd: string): HeroActionsType;
export declare function removeSkill(skillToRemove: string): HeroActionsType;
export declare function clearAddHero(): {
    type: string;
};
export declare function setEmail(email: string): HeroActionsType;
export declare function setPassword(password: string): HeroActionsType;
export declare function setRegEmail(regEmail: string): HeroActionsType;
export declare function setRegPassword(regPassword: string): HeroActionsType;
export declare function setUserId(userId: string): HeroActionsType;
export declare function setJWT(jwt: string): HeroActionsType;
export declare function logout(): HeroActionsType;
export declare function setUserName(userName: string): HeroActionsType;
export declare function setUserSurname(surname: string): HeroActionsType;
export declare function setUserPhone(phone: string): HeroActionsType;
export declare function clearRegData(): HeroActionsType;
