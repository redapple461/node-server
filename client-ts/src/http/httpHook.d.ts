import { Hero } from '../models/Hero';
export declare const getHeroes: (token: string) => Promise<any>;
export declare const addHero: (data: Hero, token: string) => Promise<any>;
export declare const getByName: (name: string, token: string) => Promise<any>;
export declare const deleteByName: (name: string, token: string) => Promise<Response>;
export declare const updateHero: (oldName: string, body: Hero, token: string) => Promise<any>;
export declare const signIn: (email: string, password: string) => Promise<any>;
export declare const register: (email: string, password: string, name: string, surname: string, phone: string) => Promise<any>;
