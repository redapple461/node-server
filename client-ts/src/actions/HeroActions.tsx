export const SET_MARVEL = 'SET_MARVEL';
export const SET_DC = 'SET_DC';
export const SET_BOTH = 'SET_BOTH';
export const GET_DATA = 'GET_DATA';
export const HEROES_LOADED = 'HEROES_LOADED';
export const ADD_HERO = 'ADD_HERO';
export const DELETE_HERO = 'DELETE_HERO';
export const INIT_HERO = 'INIT_HERO';
export const UPDATE_NAME = 'UPDATE_NAME';
export const UPDATE_ADD_HERO_NAME = 'UPDATE_ADD_HERO_NAME';
export const UPDATE_ADD_HERO_UNIVERSE = 'UPDATE_ADD_HERO_UNIVERSE';
export const CLEAR_DETAIL_HERO = 'CLEAR_DETAIL_HERO';
export const DETAIL_LOADED = 'DETAIL_LOADED';
export const UPDATE_HERO = 'UPDATE_HERO';
export const UPDATE_UNIVERSE = 'UPDATE_UNIVERSE';
