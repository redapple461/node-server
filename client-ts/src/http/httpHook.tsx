import {Hero} from '../models/Hero';
import {User} from '../models/User';

export const getHeroes = (token: string) => {
	return fetch('http://localhost:4000/getHeroes',{
		method: 'GET',
		headers: {
			'Content-Type': 'application/json',
			Authorization: token
		}
	})
		.then(res => res.json());
};

export const addHero = (data: Hero, token: string) => {
	return fetch('http://localhost:4000/addHero', {
		method: 'POST',
		body: JSON.stringify(data),
		headers: {
			'Content-Type': 'application/json',
			Authorization: token
		}
	}).then(res => res.json());
};

export const getByName = (name: string, token: string) => {
	return fetch('http://localhost:4000/getHero/' + name, {
		method: 'GET',
		headers: {
			'Content-Type': 'application/json',
			Authorization: token
		}
	}).then(res => res.json());
};

export const deleteByName = (name: string, token: string) => {
	return fetch('http://localhost:4000/deleteHero/' + name, {
		method: 'DELETE',
		headers: {
			Authorization: token
		}
	});
};

export const updateHero = (oldName: string, body: Hero,token: string) => {
	return fetch('http://localhost:4000/updateHero/' + oldName, {
		method: 'PUT',
		body: JSON.stringify(body),
		headers: {
			'Content-Type': 'application/json',
			Authorization: token
		}
	}).then(res => res.json());
};

export const signIn = (email: string, password: string) => {
	return fetch('http://localhost:4000/auth/login/', {
		method: 'POST',
		body: JSON.stringify({
			email,
			password
		}),
		headers: {
			'Content-Type': 'application/json'
		}
	}).then(res => res.json());
};

export const register =(email: string, password: string, name: string, surname: string, phone: string) => {
	return fetch('http://localhost:4000/auth/register', {
		method: 'POST',
		body: JSON.stringify({
			email,
			password,
			name,
			surname,
			phone
		}),
		headers: {
			'Content-Type': 'application/json',
		}
	}).then(res => res.json());
};

export const change = (id: string, token: string, user: User) => {
	return fetch(`http://localhost:4000/user/getUser/:${id}`, {
		method: 'GET',
		body: JSON.stringify(user),
		headers: {
			'Content-Type': 'application/json',
			Authorization: token
		}
	});
}