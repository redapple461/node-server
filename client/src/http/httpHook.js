


export const getHeroes = () => {
    return fetch("http://localhost:4000/getHeroes")
        .then(res => res.json());
}


export const addHero = (data) => {
    return fetch("http://localhost:4000/addHero", {
                method: 'POST', 
                body: JSON.stringify(data),
                headers: {
                  'Content-Type': 'application/json'
                }
            }).then(res => res.json());
}

export const getByName = (name) => {
    return fetch('http://localhost:4000/getHero/'+name)
        .then(res => res.json());
}

export const deleteByName = (name) => {
    return fetch("http://localhost:4000/deleteHero/"+name,{
        method: 'DELETE'
   });
}

export const updateHero = (oldName, body) => {
    return fetch('http://localhost:4000/updateHero/'+oldName,{
                method: 'PUT',
                body: JSON.stringify(body),
                headers:{
                    'Content-Type': 'application/json'
                }
            }).then(res => res.json())
}