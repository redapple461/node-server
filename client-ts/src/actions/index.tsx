import { Hero } from '../models/Hero';
import {HeroActionsType} from '../actions/types/HeroTypes';
import * as actions from '../actions/HeroActions';

export function setMarvel (): HeroActionsType {
	return{
		type: actions.SET_MARVEL
	};
}

export  function setDC (): HeroActionsType {
	return{
		type: actions.SET_DC
	};
}

export function setBoth (): HeroActionsType{
	return{
		type: actions.SET_BOTH
	};
}

export function getData (data: Hero[]): HeroActionsType {
	return{
		type: actions.GET_DATA,
		data: data
	};
}

export function loadComplete(): HeroActionsType {
	return{
		type: actions.HEROES_LOADED
	};
}

export function addHero(hero: Hero): HeroActionsType {
	return{
		type: actions.ADD_HERO,
		hero: hero
	};
}

export function deleteHero(id: number): HeroActionsType{
	return{
		type: actions.DELETE_HERO,
		id: id
	};
}

export function initDetailHero(hero: Hero): HeroActionsType {
	return{
		type: actions.INIT_HERO,
		hero: hero
	};
}

export function updateName(name: string): HeroActionsType {
	return{
		type: actions.UPDATE_NAME,
		newName: name
	};
}

export function updateUniverse(universe: string): HeroActionsType{
	return{
		type: actions.UPDATE_UNIVERSE,
		newUniverse: universe
	};
}

export function updateHero(): HeroActionsType {
	return{
		type: actions.UPDATE_HERO
	};
}

export function updateAddHeroName(name: string): HeroActionsType{
	return{
		type: actions.UPDATE_ADD_HERO_NAME,
		addName: name,
	};
}

export function updateAddHeroUniverse(universe: string): HeroActionsType{
	return{
		type: actions.UPDATE_ADD_HERO_UNIVERSE,
		addUniverse: universe
	};
}

export function clearDetailHero(): HeroActionsType{
	return{
		type: actions.CLEAR_DETAIL_HERO
	};
}

export function isLoad(): HeroActionsType {
	return{
		type: actions.DETAIL_LOADED
	};
}
