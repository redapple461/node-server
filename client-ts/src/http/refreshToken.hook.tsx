import { updateToken } from './httpHook';
import * as actions from '../actions';

export const refreshToken = async (dispatch) => {

	// alert('try to update token');
	const refToken = JSON.parse(localStorage.getItem('user_data')).refreshToken;
	// const dispatch = useDispatch();
	// console.log('refresh '+refreshToken);
	// console.log('auth '+JSON.parse(localStorage.getItem('user_data')).token);
	return updateToken(refToken).then(async _res => {
		// console.log(_res);
		const newUserData = (JSON.parse(localStorage.getItem('user_data')));
		newUserData.token = _res.token;
		dispatch(actions.setJWT(_res.token));
		// console.log('test '+token);
		localStorage.setItem('user_data', JSON.stringify(newUserData));
		window.M.toast({html: 'Token updated'});
		// console.log('recursia');
	});
};
