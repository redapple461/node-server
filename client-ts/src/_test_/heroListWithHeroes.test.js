import React from 'react';
import HeroList from '../pages/heroList'
import { mount,shallow } from 'enzyme';
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { Switch,Route } from 'react-router-dom';
import {BrowserRouter as Router} from 'react-router-dom'
import * as actions from '../actions'
import "babel-polyfill"
import 'jest-fetch-mock'
import { deleteByName } from '../http/httpHook'

describe('<HeroList/> test', () => {
    let store;
    let container;
    const mockStore = configureStore([]);


    beforeEach(() => {
        store = mockStore({
            searchUniverse: "",
            heroes: [
                {id:0, name:'Mock_1', universe:'Mock_U'}
            ],
            oldName: "",
            addHero: {name: "Mock_ADD", universe: "Mock_AU"},
            detailHero: null,
            isLoad: false,
            noHeroes: false
        });
        //fetch = jest.fn(() => {body: 'ok'});
        store.dispatch = jest.fn();
        container = renderer.create(
            <Provider store={store}>
              <Router>
                <Route>
                    <HeroList/>
                </Route>
              </Router>
            </Provider>    
        );
    })

    it('should render with given state from Redux store', done => {
        // console.log(container.toJSON().children[3].children[2].props)
        expect(container.toJSON()).toMatchSnapshot();
        done();
    });

    it('should dispatch after add button clicked', done => {
        renderer.act( () => {
            container.toJSON().children[3].children[2].props.onClick()
        });
        expect(store.dispatch).toHaveBeenCalledTimes(0);
       // expect(store.dispatch).toHaveBeenCalledWith(
       //     {type:'HEROES_LOADED'}
       // )
        done();
    })
    
    it('should exist after click on delete btn', done => {
        const btn = container.toJSON().children[2].children[0].children[1];
        renderer.act(() => {
            btn.props.onClick()
        });
        expect(btn.children[1].trim()).toBe('x');
        done();
    })

})