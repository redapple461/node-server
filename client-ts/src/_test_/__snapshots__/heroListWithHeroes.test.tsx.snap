// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeroList/> test should render with given state from Redux store 1`] = `
Array [
  <h1>
     Heroes List
  </h1>,
  <a
    href="/main"
    onClick={[Function]}
  >
     
    <button
      className="waves-effect waves-light btn"
    >
       
      Dashboard
       
    </button>
  </a>,
  <ul>
    <div>
      <a
        href="/detailHero/Mock_1"
        onClick={[Function]}
      >
        <button
          className="waves-effect orange darken-1 btn list_btn"
        >
           
          0: Mock_1
           
        </button>
      </a>
      <button
        className="waves-effect red btn"
        onClick={[Function]}
      >
         
        x
         
      </button>
    </div>
  </ul>,
  <div>
    Add new hero
    <input
      id="add_input"
      onChange={[Function]}
      type="text"
    />
    <button
      className="waves-effect waves-light btn"
      id="add_btn"
      onClick={[Function]}
    >
       
      Add hero
       
    </button>
    <div>
      <strong>
         Choose universe: 
      </strong>
      <label>
        <input
          className="with-gap"
          id="rMarvel"
          name="group1"
          onChange={[Function]}
          type="radio"
          value="Marvel"
        />
        <span>
          Marvel
        </span>
      </label>
      <label>
        <input
          className="with-gap"
          id="rDC"
          name="group1"
          onChange={[Function]}
          type="radio"
          value="DC"
        />
        <span>
          DC
        </span>
      </label>
    </div>
    <div>
      <strong>
         Choose skills: 
      </strong>
      <label>
        <input
          className="with-gap"
          id="power_hit"
          onChange={[Function]}
          type="checkbox"
          value="Power Hit"
        />
        <span>
          Power Hit
        </span>
      </label>
      <label>
        <input
          className="with-gap"
          id="heal"
          onChange={[Function]}
          type="checkbox"
          value="Heal"
        />
        <span>
          Heal
        </span>
      </label>
    </div>
  </div>,
  <button
    className="waves-effect waves-light btn"
    onClick={[Function]}
  >
     
    Go back
     
  </button>,
]
`;
